module/mempool/herocache/dns_cache.go

```diff
@@ -0,0 +1,143 @@
+package herocache
+
+import (
+	"net"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+)
+
+type DNSCache struct {
+	ipCache  *stdmap.Backend
+	txtCache *stdmap.Backend
+}
+
+func NewDNSCache(sizeLimit uint32, logger zerolog.Logger) *DNSCache {
+	return &DNSCache{
+		txtCache: stdmap.NewBackendWithBackData(
+			herocache.NewCache(sizeLimit, 8, heropool.LRUEjection, logger)),
+		ipCache: stdmap.NewBackendWithBackData(
+			herocache.NewCache(sizeLimit, 8, heropool.LRUEjection, logger)),
+	}
+}
+
+// PutIpDomain adds the given ip domain into the cache.
+func (d *DNSCache) PutIpDomain(domain string, timestamp int64, addresses []net.IPAddr) bool {
```

Comment:
Any reason why this is called `PutIpDomain()` and not `PutDomainIp()`?
Also the argument order is 1st domain, 2nd timestamp, 3rd addresses.
If the function does end up getting renamed to `PutDomainIp()`, why not use the argument order 1st domain, 2nd addresses, 3rd timestamp... to sort of match the name?

Reply:
https://github.com/onflow/flow-go/pull/1888/commits/a0728490ee1136453bf6e1102f2db006d8d73e6c